### PR TITLE
table csv download

### LINKED ISSUE #427: Possibility to download a Taipy table as a CSV file
**What would that feature address**
It is common to want to be able to download a CSV. It will facilitate if someone could download a Taipy table directly as CSV.

***Description of the ideal solution***
A new property called `downloadable` could be created. When `True`, a button somewhere could be clicked on to download the CSV.

**Acceptance Criteria**
- [ ] Ensure new code is unit tested, and check code coverage is at least 90%
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change
- [ ] Ensure any change is well documented

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
add a downloadable boolean property to table

resolves #427

### CODE REVIEW

File: taipy/gui/gui.py [hunk 1]
```diff
@@ -1320,14 +1323,47 @@ def _get_user_instance(self, class_name: str, class_type: type) -> t.Union[objec
...
 
+    def __download_csv(self, state: State, var_name: str, payload: dict):
+        holder_name = t.cast(str, payload.get("var_name"))
+        ret = self._accessors._get_data(
+            self,
+            holder_name,
+            _getscopeattr(self, holder_name, None),
+            {"alldata": True, "csv": True},
+        )
+        if isinstance(ret, dict):
```
Comment (dinhlongviolin1) [inline taipy/gui/gui.py:1334]: Could it be a _MapDict

**Review by dinhlongviolin1 — APPROVED** [review]
LGTM

File: taipy/gui/gui.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/gui.py:1334]: no, it comes from get_data

**Review by FabienLelaquais — APPROVED** [review]
Nice!

**Pull request merged**